app/helpers/application_helper.rb

```diff
@@ -84,6 +84,10 @@ def content_preview_url(manual)
     "#{Plek.external_url_for('draft-origin')}/#{manual.slug}"
   end
 
+  def allow_publish?(manual, slug_unique)
```

Comment:
Since we can derive slug_unique within this method (by calling `manual.slug_unique?(current_user)`) could the slug_unique argument be superfluous?

Though my subsequent comment may render this irrelevent.

Reply:
I considered this, but wasn't sure about having the view call a method that is ultimately doing a DB query. The code is basically just moved as-is from the show view, which is another reason I did it this way.

Reply:
Sure I can see your logic - we don't tend to treat methods as special if they call the DB and tend to have a bunch of methods hit the DB at rendering time (particularly with some of Rails' lazy loading). So in general, if we'd be concerned about performance we'd likely do memoising in the method to ensure queries aren't made multiple times.

Reply:
Fair enough.

Reply:
We've changed this to use an optional parameter so as not to change how the existing show view was working (in case there was a good reason for that to be doing the check once and passing the value around).

For the `confirm_publish` controller method, it calls it without passing a second argument, and the helper will default to calling the method on the manual.

Reply:
Cool - that seems a neat compromise. I do think though the code ends up a bit heavy on the wtf factor having such an involved default argument so I think it'd be better to just passed in in both places. 

Sorry about the to and fro.

Reply:
Yeah, that's fair. Have now reverted the parameter back to being required and passing the argument value in both places.